compiler/src/dmd/semantic2.d

```diff
@@ -819,3 +820,55 @@ private void doGNUABITagSemantic(ref Expression e, ref Expression* lastTag)
     }
     ale.elements.sort!predicate;
 }
+
+/**
+ * Try lower a variable's static Associative Array to a newaa struct.
+ * Params:
+ *   vd = Variable to lower
+ *   sc = Scope
+ */
+private void tryLowerStaticAA(VarDeclaration vd, Scope* sc)
+{
+    if (!vd.isDataseg())
+        return;
+    if (auto ei = vd._init.isExpInitializer())
+    {
+        scope v = new StaticAAVisitor(sc);
+        ei.exp.accept(v);
+    }
+}
+
+/// Visit Associative Array literals and lower them to structs for static initialization
+private extern(C++) final class StaticAAVisitor : SemanticTimeTransitiveVisitor
+{
+    alias visit = SemanticTimeTransitiveVisitor.visit;
+    Scope* sc;
+
+    this(Scope* sc) scope @safe
+    {
+        this.sc = sc;
+    }
+
+    override void visit(AssocArrayLiteralExp initExp)
+    {
+        if (!verifyHookExist(initExp.loc, *sc, Id._aaAsStruct, "initializing static associative arrays", Id.object))
+            return;
+
+        Expression exp = initExp.syntaxCopy();
+        Expression id = new IdentifierExp(exp.loc, Id.empty);
+        id = new DotIdExp(exp.loc, id, Id.object);
+        id = new DotIdExp(exp.loc, id, Id._aaAsStruct);
+        auto arguments = new Expressions();
+        arguments.push(exp.syntaxCopy());
```

Comment:
This second syntax copy does not look necessary, `exp` is a fresh copy of `initExp` and it gets reassigned anyway.

Reply:
Good catch. The code was hastily copy pasted from other lowerings, so I've also given better variable names now.